tools/esi/esi-cosim-runner.py.in

```diff
@@ -25,287 +25,284 @@ ThisFileDir = os.path.dirname(__file__)
-        # installed and working.
-        cmd = [self.simRunScript, "--no-run",
-               "--objdir", "o"] \
-            + self.sources + self.args
-        print("[INFO] Compile command: " + " ".join(cmd))
-        vrun = subprocess.run(
-            cmd,
-            capture_output=True,
-            text=True)
-        # cwd=self.execdir)
-        output = vrun.stdout + "\n----- STDERR ------\n" + vrun.stderr
-        if vrun.returncode != 0:
-            print("====== Compilation failure:")
-            print(output)
-        print(f"[INFO] Compile time: {time.time()-start}")
-        return vrun.returncode
-
-    def writeScript(self, port):
-        """Write out the test script."""
-
-        with open("script.py", "w") as script:
-            # Include a bunch of config variables at the beginning of the
-            # script for use by the test code.
-            vars = {
-                "srcdir": self.srcdir,
-                "srcfile": self.file,
-                # 'rpcSchemaPath' points to the CapnProto schema for RPC and is
-                # the one that nearly all scripts are going to need.
-                "rpcschemapath": self.schema
-            }
-            script.writelines(f"{name} = \"{value}\"\n" for (
-                name, value) in vars.items())
-            script.write("\n\n")
-
-            # Add the test files directory and this files directory to the
-            # pythonpath.
-            script.write("import os\n")
-            script.write("import sys\n")
-            script.write(
-                f"sys.path.append(\"{os.path.dirname(self.file)}\")\n")
-            script.write(
-                f"sys.path.append(\"{os.path.dirname(__file__)}\")\n")
-            script.write("\n\n")
-            script.write(
-                "simhostport = f'{os.uname()[1]}:" + str(port) + "'\n")
-
-            # Run the lines specified in the test file.
-            script.writelines(
-                f"{x}\n" for x in self.runs)
-
-    def run(self):
-        """Run the test by creating a Python script, starting the simulation,
+    self.args = addlArgs
+    self.file = testFile
+    self.runs = list()
+    self.srcdir = os.path.dirname(self.file)
+    self.sources = list()
+    self.top = "top"
+
+    if "@ESI_COSIM_PATH@" == "" or not os.path.exists("@ESI_COSIM_PATH@"):
+      raise Exception("The ESI cosimulation DPI library must be " +
+                      "enabled to run cosim tests.")
+
+    self.simRunScript = os.path.join("@CIRCT_TOOLS_DIR@", "circt-rtl-sim.py")
+
+    if schema == "":
+      schema = os.path.join("@CIRCT_MAIN_INCLUDE_DIR@", "circt", "Dialect",
+                            "ESI", "cosim", "CosimDpi.capnp")
+    self.schema = schema
+
+    fileReader = open(self.file, "r")
+    sources = []
+    for line in fileReader:
+      # Arguments to circt-rtl-sim, except for source files list
+      m = re.match(r"^//\s*ARGS:(.*)$", line)
```

Comment:
This, and the similar lines below, are the only non-format changes. I removed the use of `m := ...` because the `:=` operator is only available after Python 3.8. I have no problem requiring that version myself, but it seems like this was the only change required to support an older Python 3, and the added compatibility seems to outweigh the benefit of `:=` here.

The reason I am even touching this is because our CI machine doesn't come with Python 3.8, and it seemed preferable to increase the compatibility of this chunk of code rather than using the latest Python in CI.

Reply:
Agreed

Reply:
:+1: thanks. I set up ESI cosim tests locally to make sure this still worked, and they all pass.